### DISCUSSION TITLE
Chumsky Zero Copy Release Date

### DISCUSSION THREAD

**Discussion by vrindisbacher** [zesterer/chumsky#455] [Q&A] [answered]
I want to start off by saying that Chumsky is an amazing tool. Most of my experience with parser implementations come from OCaml's Menhir, which is a good tool but has a steep learning curve. With Chumsky, one of my friends read the code and understood exaclty what was happening. Please please please, keep up the great work. 

I saw that the Zero Copy version of Chumsky is incredibly performant. I'd like to leverage this asap - but I see that the release is only in beta. Is there an estimate for when the release will be out of beta and stable?

**Comment by zesterer** [comment, marked as answer]
Hello!

Chumsky is a hobby project for me so it moves forward as and when I have the time and mental energy to move it forward. I find that this comes in fits and bursts: sometimes a month's worth of work gets done in a few days, sometimes a few day's worth of work gets done in a month. That said, 1.0 is getting very close to release. There are only a few things left on the list beforehand. Roughly speaking, they are:

- Finish adding documentation to new features, fix inaccuracies in existing documentation
- Finish off the guide to improve the experience people have with the crate as much as possible
- Take a deeper look at #425 and see whether the API can be made more flexible
- Do a final pass over the existing API to ensure that there's nothing that might create problems in the future

You'll notice that there's nothing in here that's particularly radical: you can start building on top of the alpha versions of 1.0 today and, in all likelihood, very little (or perhaps even nothing) will break between now and the stable 1.0 release. If you want to get started with the crate, I can definitely recommend whacking `chumsky = "1.0-alpha.4"` into your `Cargo.toml` and getting stuck in, the only issues you should encounter should be occasionally out of date documentation.

Unfortunately I can't give a hard number on the release date, but it's definitely quite close now.

**Reply by vrindisbacher** [reply]
Thanks for such a detailed response! This sounds great. I will likely plug into the alpha version.
